railgun/config/base/kustomization.yaml

```diff
@@ -0,0 +1,15 @@
+resources:
```

Comment:
Do you plan to consider maximize usage of files under deploy/manifests, deploy/single to remove duplicates within this PR ?

Reply:
Yes, hence why, for example, ` railgun/config/base/kong-ingress-dbless.yaml` is just a symlink to ` ../../../deploy/manifests/base/kong-ingress-dbless.yaml `. That's a bit clearer on the filesystem than it is in Github's UI:

```
$ ls railgun/config/base -l
total 16
-rw-r--r-- 1 traines traines  298 Jul 21 10:30 kong-ingress-2x.yaml
lrwxrwxrwx 1 traines traines   55 Jul 15 14:54 kong-ingress-dbless.yaml -> ../../../deploy/manifests/base/kong-ingress-dbless.yaml
-rw-r--r-- 1 traines traines  316 Jul 16 11:23 kustomization.yaml
lrwxrwxrwx 1 traines traines   45 Jul 15 14:54 namespace.yaml -> ../../../deploy/manifests/base/namespace.yaml
-rw-r--r-- 1 traines traines   92 Jul 15 15:35 serviceaccount.yaml
lrwxrwxrwx 1 traines traines   43 Jul 15 14:53 service.yaml -> ../../../deploy/manifests/base/service.yaml
-rw-r--r-- 1 traines traines 1089 Jul 16 11:03 tmp-bindings.yaml
lrwxrwxrwx 1 traines traines   54 Jul 15 14:54 validation-service.yaml -> ../../../deploy/manifests/base/validation-

$ ls railgun/config/variants/postgres -l
total 8
lrwxrwxrwx 1 traines traines  64 Jul 16 11:35 kong-ingress-postgres.yaml -> ../../../../deploy/manifests/postgres/kong-ingress-postgres.yaml
-rw-r--r-- 1 traines traines 110 Jul 16 11:25 kustomization.yaml
lrwxrwxrwx 1 traines traines  52 Jul 16 11:36 migration.yaml -> ../../../../deploy/manifests/postgres/migration.yaml
lrwxrwxrwx 1 traines traines  51 Jul 16 11:36 postgres.yaml -> ../../../../deploy/manifests/postgres/postgres.yaml
```
The new content for each of the 2.x variants is kustomization.yamls (which need to be unique), patches, and the ServiceAccount (because 1.x had it included with the 1.x RBAC roles, which we don't want in 2.x).

kustomization.yaml could point to the 1.x templates directly instead, but IMO the symlink version is more readable.